Client/Frontend/Sync/BraveCore/BraveSyncAPIExtensions.swift

```diff
@@ -158,3 +158,40 @@ extension BraveSyncAPI {
         }
     }
 }
+
+extension BraveSyncAPI {
+    func getQRCodeImageV2(_ size: CGSize) -> UIImage? {
+        let hexCode = hexSeed(fromSyncCode: getSyncCode())
+        if hexCode.isEmpty {
+            return nil
+        }
+
+        // Typically QR Codes use isoLatin1, but it doesn't matter here
+        // as we're not encoding any special characters
+        guard let syncCodeData = BraveSyncQRCodeModel(syncHexCode: hexCode).jsonData,
+              !syncCodeData.isEmpty else {
+            return nil
+        }
+        
+        guard let filter = CIFilter(name: "CIQRCodeGenerator") else {
+            return nil
+        }
+
+        filter.do {
+            $0.setValue(syncCodeData, forKey: "inputMessage")
+            $0.setValue("H", forKey: "inputCorrectionLevel")
+        }
+        
+        if let image = filter.outputImage {
+            let scaleX = size.width / image.extent.size.width
+            let scaleY = size.height / image.extent.size.height
```

Comment:
Do we care about divide by 0 here?

Reply:
We don't because it should never generate an image of size zero. However, I added a check anyway.